packages/runtime/container-runtime/src/test/opLifecycle/outbox.spec.ts

```diff
@@ -341,9 +379,10 @@ describe("Outbox", () => {
 			state.batchesSubmitted.map((x) => x.messages.map((m) => m.metadata?.batchId)),
 			[
 				[undefined], // Flush 1 - ID Allocation (no batch ID used)
-				["batchId-A", undefined], // Flush 1 - Main
+				["batchId-A"], // Flush 1 - Main
 				["batchId-B"], // Flush 2 - Main
 				["batchId-C", undefined], // Flush 3 - Blob Attach
+				["batchId-C"], // Flush 3 - Blob Attach, Main empty batch
 				[undefined], // Flush 4 - Main (no batch ID given)
 			],
```

Comment:
Why is this needed now?

Reply:
Because when we flush just blob attach ops, we also flush an empty batch since main one was empty

Reply:
Fix this issue by flusing one empty batch only when both blobattach and main batch are empty on resubmit